addon/services/engine-router-service.js

```diff
@@ -0,0 +1,118 @@
+import Service from '@ember/service';
+import { assert } from '@ember/debug';
+import { action, computed, get } from '@ember/object';
+import { reads } from '@ember/object/computed';
+import { getOwner } from '@ember/application';
+import Evented from '@ember/object/evented';
+import { namespaceEngineRouteName } from '../utils/namespace-engine-route-name';
+import { getRootOwner } from '../utils/owner';
+
+export default class EngineRouterService extends Service.extend(Evented) {
+  init() {
+    super.init();
+
+    this._externalRoutes = getOwner(this)._externalRoutes;
+    this._mountPoint = getOwner(this).mountPoint;
+    this.rootApplication = getRootOwner(this);
+
+    this.externalRouter.on('routeWillChange', this.onRouteWillChange);
+    this.externalRouter.on('routeDidChange', this.onRouteDidChange);
+  }
+
+  destroy() {
+    this.externalRouter.off('routeWillChange', this.onRouteWillChange);
+    this.externalRouter.off('routeDidChange', this.onRouteDidChange);
+
+    super.destroy();
+  }
+
+  @action
+  onRouteWillChange(...args) {
+    this.trigger('routeWillChange', ...args);
+  }
+
+  @action
+  onRouteDidChange(...args) {
+    this.trigger('routeDidChange', ...args);
+  }
+
+  @reads('externalRouter.rootURL') rootURL;
+
+  @reads('externalRouter.currentURL') currentURL;
+
+  @computed('externalRouter.currentRouteName')
+  get currentRouteName() {
+    if (get(this, 'externalRouter').currentRouteName === this._mountPoint) {
```

Comment:
You can use a native accessor for `externalRouter`, like you do above already, since it's just a plain getter.

```suggestion
    if (this.externalRouter.currentRouteName === this._mountPoint) {
```